### ISSUE TITLE
 Server Crashes after a while

### ISSUE THREAD

**Issue by coxtor** [rhasspy/rhasspy-server-hermes#18]
Hi after a while of running Rhasspy, the Hermes server crashes with following error and remains nonfunctional until I restart Rhasspy.

```
[DEBUG:2020-10-18 06:04:26,553] rhasspyserver_hermes: Handling Voices (topic=rhasspy/tts/voices, id=1a5a64c8-2e2d-499b-84b8-b498d32cda87)
[ERROR:2020-10-18 06:04:46,429] rhasspyserver_hermes: 
Traceback (most recent call last):
  File "/usr/lib/rhasspy/.venv/lib/python3.7/site-packages/quart/app.py", line 1821, in full_dispatch_request
    result = await self.dispatch_request(request_context)
  File "/usr/lib/rhasspy/.venv/lib/python3.7/site-packages/quart/app.py", line 1869, in dispatch_request
    return await handler(**request_.view_args)
  File "/usr/lib/rhasspy/rhasspy-server-hermes/rhasspyserver_hermes/__main__.py", line 866, in api_listen_for_command
    handle_captured(), messages, message_types
  File "/usr/lib/rhasspy/rhasspy-server-hermes/rhasspyserver_hermes/__init__.py", line 960, in publish_wait
    result_awaitable, timeout=timeout_seconds
  File "/usr/lib/python3.7/asyncio/tasks.py", line 449, in wait_for
    raise futures.TimeoutError()
concurrent.futures._base.TimeoutError
```

**Comment by synesthesiam** [comment]
Is this with a web page open viewing it? I'm seeing some reports of issues with recurring websocket connections.

**Comment by coxtor** [comment]
It is possible, while testing I often keep multiple tabs with the Rhasspy interface open without even knowing it, as I mostly don't close the tabs. 
Can I give you further information?

**Comment by synesthesiam** [comment]
Here's the issue tracking the problem: https://github.com/rhasspy/rhasspy-server-hermes/issues/19

**Comment by coxtor** [comment]
Okay, of what I have now observed the error probably does not have anything to do with open web sockets. The issue persists with no open web interfaces.

**Comment by coxtor** [comment]
The error however is different:
```
[WARNING:2020-10-22 10:12:11,464] rhasspydialogue_hermes: No session for id default-blueberry_linux-977b4cb8-eaf0-4f63-b132-947c469e1b8d
[DEBUG:2020-10-22 10:12:15,854] rhasspyserver_hermes: <- NluIntent(input='pause', intent=Intent(intent_name='MediaPlayPause', confidence_score=1.0), site_id='default', id=None, slots=[Slot(entity='media_toggle', value={'kind': 'Unknown', 'value': 'pause'}, slot_name='media_toggle', raw_value='stopp', confidence=1.0, range=SlotRange(start=0, end=5, raw_start=0, raw_end=5))], session_id='default-blueberry_linux-05c1cccb-6f30-445f-bfbf-0b3a63363c28', custom_data=None, asr_tokens=[[AsrToken(value='pause', confidence=1.0, range_start=0, range_end=5, time=None)]], asr_confidence=None, raw_input='stopp', wakeword_id='blueberry_linux', lang=None)
[ERROR:2020-10-22 10:12:41,493] rhasspydialogue_hermes: Session timed out for site default: default-blueberry_linux-05c1cccb-6f30-445f-bfbf-0b3a63363c28
[DEBUG:2020-10-22 10:24:15,586] rhasspyserver_hermes: <- HotwordDetected(model_id='/profiles/de/porcupine/blueberry_linux.ppn', model_version='', model_type='personal', current_sensitivity=0.5, site_id='default', session_id=None, send_audio_captured=None, lang=None)
[WARNING:2020-10-22 10:24:16,136] rhasspydialogue_hermes: No session for id default-blueberry_linux-05c1cccb-6f30-445f-bfbf-0b3a63363c28
[WARNING:2020-10-22 10:24:16,138] rhasspydialogue_hermes: No session for id default-blueberry_linux-05c1cccb-6f30-445f-bfbf-0b3a63363c28
[DEBUG:2020-10-22 10:24:20,013] rhasspyserver_hermes: <- NluIntentNotRecognized(input='', site_id='default', id=None, custom_data=None, session_id='default-blueberry_linux-53396e68-e1e2-4b42-b9f0-e9bcbff45a7b')
[WARNING:2020-10-22 10:24:20,979] rhasspydialogue_hermes: No intent recognized (site_id=default, session_id=default-blueberry_linux-53396e68-e1e2-4b42-b9f0-e9bcbff45a7b)
[ERROR:2020-10-22 11:11:07,373] rhasspymicrophone_pyaudio_hermes: record
Traceback (most recent call last):
  File "/usr/lib/rhasspy/rhasspy-microphone-pyaudio-hermes/rhasspymicrophone_pyaudio_hermes/__init__.py", line 121, in record
    chunk = mic.read(self.frames_per_buffer)
  File "/usr/lib/rhasspy/.venv/lib/python3.7/site-packages/pyaudio.py", line 608, in read
    return pa.read_stream(self._stream, num_frames, exception_on_overflow)
OSError: [Errno -9981] Input overflowed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/rhasspy/rhasspy-microphone-pyaudio-hermes/rhasspymicrophone_pyaudio_hermes/__init__.py", line 128, in record
    mic.stop_stream()
  File "/usr/lib/rhasspy/.venv/lib/python3.7/site-packages/pyaudio.py", line 524, in stop_stream
    pa.stop_stream(self._stream)
OSError: Stream not open


```

**Comment by synesthesiam** [comment]
Ah, I know that error well. This happens when the CPU can't keep up with the incoming audio and it overflows. What are you running Rhasspy on?

There are some possible ways of fixing this on my end; probably the best option is to use PyAudio's callback API. Another is to expose "chunk size" parameters so you can increase how much data is pulled from the microphone on each cycle.

**Comment by coxtor** [comment]
Nice, thanks for the information - this issue makes the rhasspy unusable :( . 
I am running the client on a Raspberry Pi 4 Model B Rev 1.2 and the server runs in a dockercontainer on a asrock j3455 itx mainboard. 
The microphone is a ps3 eye cam connected directly to the mainboard, if that helps.

**Comment by synesthesiam** [comment]
I'll roll this fix into 2.5.8
